tests/runtime/out_stackdriver.c

```diff
@@ -570,6 +572,55 @@ static void cb_check_k8s_container_resource_default_regex(void *ctx, int ffd,
     flb_sds_destroy(res_data);
 }
 
+static void cb_check_trace_common_case(void *ctx, int ffd,
```

Comment:
For the reference, we have similar implementation at the Fluentd side to extract `logging.googleapis.com/trace`:

```
The value of this field should be formatted as projects/[PROJECT-ID]/traces/[TRACE-ID], so it can be used by the Logs Explorer and the Trace Viewer to group log entries and display them in line with traces. If `autoformat_stackdriver_trace` is true and [V] matches the format of ResourceTrace traceId, the LogEntry trace field has the value projects/[PROJECT-ID]/traces/[V].
```

Related documentation:
https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects.traces#Trace
https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry
https://cloud.google.com/logging/docs/agent/configuration#special-fields

Reply:
I reviewed the docs you shared and I think what we can do is to add `autoformat_stackdriver_trace` in the stackdriver output configuration. If it is set to `true`, then we can format the trace as `projects/[PROJECT-ID]/traces/[TRACE-ID]`, where `TRACE-ID` is `record['trace_key_value']`.

E.g.

Output is configured like:

```
    [OUTPUT]
        Name                    stackdriver
        Match                   stackdriver.*
        Resource              k8s_container
        Trace_Key             tag        
```
 
Then the log entry will have `trace=record['tag']`. However if we have:

```
    [OUTPUT]
        Name                                         stackdriver
        Match                                        stackdriver.*
        Resource                                   k8s_container
        Trace_Key                                  tag      
        Autoformat_Stackdriver_Trace true
```

then `trace=projects/[PROJECT-ID]/traces/[TRACE-ID]` and `[TRACE-ID]=record['tag']`

Is this what you suggest?

Currently we have short traces without project id and then web requests are coming from istio where the trace is also short and stackdriver properly aggregates everything. So we don't necessarily want to have the full trace format.

What do you think?

Reply:
Sounds good to me (Sorry I somehow missed the email notification).

Reply:
Done! @qingling128 can you take a look? This PR has been open for quite a long time, I'll be very grateful and happy if we can close it soon.

Reply:
Just LGTM'd